crates/edr_macros/src/console_fmt.rs

```diff
@@ -19,38 +21,36 @@ pub fn console_fmt(input: &DeriveInput) -> TokenStream {
 fn derive_struct(s: &DataStruct) -> TokenStream {
     let imp = impl_struct(s).unwrap_or_else(|| quote!(String::new()));
     quote! {
-        fn fmt(&self, _spec: crate::abi::fmt::FormatSpec) -> String {
+        fn fmt(&self, _spec: edr_common::fmt::FormatSpec) -> String {
             #imp
         }
     }
 }
 
 fn impl_struct(s: &DataStruct) -> Option<TokenStream> {
-    let fields = match &s.fields {
-        Fields::Named(fields) => &fields.named,
-        Fields::Unnamed(fields) => &fields.unnamed,
-        Fields::Unit => return None,
-    };
+    if s.fields.is_empty() {
+        return None;
+    }
 
-    if fields.is_empty() {
+    if matches!(s.fields, Fields::Unit) {
         return None;
     }
 
+    let fields = s.fields.iter().collect::<Vec<_>>();
     let first_ty = match &fields.first().unwrap().ty {
         Type::Path(path) => path.path.segments.last().unwrap().ident.to_string(),
         _ => String::new(),
     };
 
-    let args: Punctuated<TokenStream, Token![,]> = fields
+    let members = s.fields.members().collect::<Vec<_>>();
```

Comment:
similarly here, we don't need to create a new `Vec` here, we can use `s.fields.members()` directly in the following line 

```rust
let args: Punctuated<TokenStream, Token![,]> = s.fields.members()
        .into_iter()
        .map(
        // ...
...
```